src/python_testing/matter_testing_infrastructure/chip/testing/spec_parsing.py

```diff
@@ -518,50 +519,75 @@ class DataModelLevel(str, Enum):
     kDeviceType = 'device_types'
 
 
-def _get_data_model_root() -> str:
-    """Attempts to find ${CHIP_ROOT}/data_model or equivalent."""
+def _get_data_model_root() -> pathlib.PosixPath:
+    """Attempts to find the 'data_model' directory inside the 'chip.testing' package."""
+    try:
+        # Locate the directory where the 'chip.testing' module is located
+        package_dir = pathlib.Path(chip.testing.__file__).parent
 
-    # Since this class is generally in a module, we have to rely on being bootstrapped or
-    # we use CWD if we cannot
-    choices = [os.getcwd()]
+        # Construct the path to the 'data_model' directory (assuming this structure)
+        data_model_root = package_dir / 'data_model'
 
-    if 'PW_PROJECT_ROOT' in os.environ:
-        choices.insert(0, os.environ['PW_PROJECT_ROOT'])
+        if not data_model_root.exists():
+            raise FileNotFoundError(f"Data model directory not found in the package at {data_model_root}.")
 
-    for c in choices:
-        data_model_path = os.path.join(c, 'data_model')
-        if os.path.exists(os.path.join(data_model_path, 'master', 'scraper_version')):
-            return data_model_path
-    raise FileNotFoundError('Cannot find a CHIP_ROOT/data_model path. Tried %r as prefixes.' % choices)
+        return data_model_root
 
+    except Exception as e:
+        raise FileNotFoundError(f"Failed to find the data model root: {e}")
 
-def get_data_model_directory(data_model_directory: typing.Union[PrebuiltDataModelDirectory, str], data_model_level: DataModelLevel) -> str:
-    if data_model_directory == PrebuiltDataModelDirectory.k1_3:
-        return os.path.join(_get_data_model_root(), '1.3', data_model_level)
-    elif data_model_directory == PrebuiltDataModelDirectory.k1_4:
-        return os.path.join(_get_data_model_root(), '1.4', data_model_level)
-    elif data_model_directory == PrebuiltDataModelDirectory.kMaster:
-        return os.path.join(_get_data_model_root(), 'master', data_model_level)
+
+def get_data_model_directory(data_model_directory: Union[PrebuiltDataModelDirectory, str], data_model_level: str) -> pathlib.PosixPath:
+    """
+    Get the directory of the data model for a specific version and level from the installed package.
+    """
+    data_model_root = _get_data_model_root()
+
+    # Build path based on the version and data model level
+    if isinstance(data_model_directory, PrebuiltDataModelDirectory):
+        version_map = {
+            PrebuiltDataModelDirectory.k1_3: '1.3',
+            PrebuiltDataModelDirectory.k1_4: '1.4',
+            PrebuiltDataModelDirectory.kMaster: 'master',
+        }
+
+        version = version_map.get(data_model_directory)
+        if not version:
+            raise ValueError(f"Unsupported data model directory: {data_model_directory}")
+
+        return data_model_root / version / data_model_level
     else:
-        return data_model_directory
+        # If it's a custom directory, return it directly
```

Comment:
why is this branch needed? This seems to effectively bypass any functionality in this method. Why would the caller not know what it needs?